### PR TITLE
Kwargs and return_value handled in Accessor

### PR DESCRIPTION
**Pull request by sachintendulkar576123**
## Proposed changes

This PR:-

- We are handling kwargs and return_value in metamodel accessor 

## Types of changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/monocle2ai/monocle/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kshitiz-okahu** [comment]
Here are a few initial thoughts:

1. The number of arguments, if possible, should be lowered. Maybe just pass a dictionary with return_value, instance etc.
2. Can we also shift the data.input to the output processor, this should be possible now that the return value is available in output processor

File: src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json [hunk 1]
```diff
@@ -5,30 +5,30 @@
       {
         "_comment": "provider type ,name , deployment , inference_endpoint",
         "attribute": "type",
-        "accessor": "lambda instance,args:'inference.azure_oai'"
+        "accessor": "lambda instance,args,kwargs,return_value:'inference.azure_oai'"
```
Comment (oi-raanne) [inline src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json]: the names should be easily comprehended

**Review by oi-raanne — CHANGES_REQUESTED** [review]

File: src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json [hunk 2]
```diff
@@ -5,30 +5,30 @@
       {
         "_comment": "provider type ,name , deployment , inference_endpoint",
         "attribute": "type",
-        "accessor": "lambda instance,args:'inference.azure_oai'"
+        "accessor": "lambda arguments:'inference.azure_oai'"
       },
       {
         "attribute": "provider_name",
-        "accessor": "lambda instance,args:args['provider_name']"
+        "accessor": "lambda arguments:arguments['kwargs']['provider_name']"
```
Comment (oi-raanne) [inline src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json:12]: add data type to arguments

**Review by kshitiz-okahu — APPROVED** [review]

File: src/monocle_apptrace/wrap_common.py [hunk 3]
```diff
@@ -164,7 +164,8 @@ def process_span(to_wrap, span, instance, args):
                         if attribute and accessor:
                             attribute_name = f"entity.{span_index}.{attribute}"
                             try:
-                                result = eval(accessor)(instance, args)
+                                arguments = {"instance":instance, "args":args, "kwargs":kwargs, "return_value":return_value}
```
Comment (oi-raanne) [inline src/monocle_apptrace/wrap_common.py:167]: instead of 'return_value' can it be 'result' or 'output'

**Review by oi-raanne — CHANGES_REQUESTED** [review]

File: src/monocle_apptrace/wrap_common.py (hunk 3 above)
Comment (oi-raanne) [reply src/monocle_apptrace/wrap_common.py:167]: also, document what goes in arguments in [monocle_user_guide.md](https://github.com/monocle2ai/monocle/blob/main/Monocle_User_Guide.md)

**Review by oi-raanne — CHANGES_REQUESTED** [review]

File: src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json (hunk 2 above)
Comment (sachintendulkar576123) [reply src/monocle_apptrace/metamodel/maps/attributes/inference/langchain_entities.json:12]: Not possible to write data type to arguments in lambda, https://stackoverflow.com/questions/33833881/is-it-possible-to-type-hint-a-lambda-function

File: Monocle_User_Guide.md [hunk 4]
```diff
@@ -40,6 +40,15 @@ You need to import monocle package and invoke the API ``setup_monocle_telemetry(
 
 ### Using Monocle's out of box support of genAI technology components
 Monocle community has done the hard work of figuring out what to trace and how to extract relevant details from multiple genAI technology components. For example, if you have a python app coded using LlamaIndex and using models hostsed in OpenAI, Monocle can seamlessly trace your app. All you need to do enable Monocle tracing.
+
+### Using Monocle's User Support for Adding Custom Attributes
```
Comment (oi-raanne) [inline Monocle_User_Guide.md:44]: remove 'User' in the above heading

File: docs/design.md [hunk 5]
```diff
@@ -51,6 +51,48 @@ The wrapper then performs the following tasks:
 - extracts the llm response from the **llm_response** value returned.
 
 
+## How to user add custom attributes in monocle's
```
Comment (oi-raanne) [inline docs/design.md:54]: title can be  'how user can add custom attributes in Monocle'

**Review by oi-raanne — CHANGES_REQUESTED** [review]

**Review by oi-raanne — APPROVED** [review]

**Pull request merged**